crates/iceberg/src/arrow/reader.rs

```diff
@@ -186,4 +216,399 @@ impl ArrowReader {
             Ok(ProjectionMask::leaves(parquet_schema, indices))
         }
     }
+
+    fn get_row_filter(&self, parquet_schema: &SchemaDescriptor) -> Result<Option<RowFilter>> {
+        if let Some(predicates) = &self.predicates {
+            let field_id_map = self.build_field_id_map(parquet_schema)?;
+
+            // Collect Parquet column indices from field ids
+            let mut collector = CollectFieldIdVisitor { field_ids: vec![] };
+            visit_predicate(&mut collector, predicates).unwrap();
```

Comment:
We don't need to do this for every file?

Reply:
Hmm, but we can only get `SchemaDescriptor` from the parquet reader for the each file.

Reply:
Oh, I got your point. I moved `CollectFieldIdVisitor` out of the loop of `tasks`.